corporate/lib/analytics.py

```diff
@@ -60,25 +57,59 @@ def estimate_annual_recurring_revenue_by_realm() -> Dict[str, int]:  # nocoverag
     return annual_revenue
 
 
+def estimate_remote_entity_annual_revenue(
+    plan: CustomerPlan, ledger_entry: Optional[LicenseLedger]
+) -> int:
+    if plan.tier in (
+        CustomerPlan.TIER_SELF_HOSTED_BASE,
+        CustomerPlan.TIER_SELF_HOSTED_LEGACY,
+        CustomerPlan.TIER_SELF_HOSTED_COMMUNITY,
+    ):
+        # None of these plan tiers generate revenue
+        renewal_cents = 0
+    elif plan.fixed_price is not None:
+        renewal_cents = plan.fixed_price
+    elif ledger_entry is None:  # nocoverage
+        renewal_cents = 0
+    elif ledger_entry.licenses_at_next_renewal is None:  # nocoverage
+        renewal_cents = 0
```

Comment:
These cases might deserve comments for when we expect them to occur, to help a reader understand which cases we're not testing?

Also, these functions should probably just be `return 0` rather than setting a variable ; doing so would let you cut the `else` at the end.

Reply:
So, I was basically recreating the get_customer_plan_renewal_amount function in the BillingSession class. And that didn't make sense, so I went back and revised this to update that function so that we always pass a LicenseLedger object along with the CustomerPlan. That takes out a lot of the no coverage cases.